java/ql/lib/ext/java.lang.model.yml

```diff
@@ -88,33 +99,45 @@ extensions:
       - ["java.lang", "String", False, "valueOf", "(char[])", "", "Argument[0]", "ReturnValue", "taint", "manual"]
       - ["java.lang", "String", False, "valueOf", "(char[],int,int)", "", "Argument[0]", "ReturnValue", "taint", "manual"]
       - ["java.lang", "String", False, "valueOf", "(int)", "", "Argument[0]", "ReturnValue", "taint", "manual"]
+      - ["java.lang", "String", False, "valueOf", "(long)", "", "Argument[0]", "ReturnValue", "taint", "manual"]
       - ["java.lang", "StringBuffer", True, "StringBuffer", "(CharSequence)", "", "Argument[0]", "Argument[-1]", "taint", "manual"]
       - ["java.lang", "StringBuffer", True, "StringBuffer", "(String)", "", "Argument[0]", "Argument[-1]", "taint", "manual"]
       - ["java.lang", "StringBuilder", True, "StringBuilder", "", "", "Argument[0]", "Argument[-1]", "taint", "manual"]
       - ["java.lang", "System", False, "arraycopy", "", "", "Argument[0]", "Argument[2]", "taint", "manual"]
+      - ["java.lang", "System", False, "getProperty", "(String)", "", "Argument[-1].MapValue", "ReturnValue", "value", "manual"]
```

Comment:
Isn't `getProperty` a static method in the `System` class?
I would have expected `getProperty : Argument[0] -> ReturnValue`

Reply:
> Isn't `getProperty` a static method in the `System` class?

yes

> I would have expected `getProperty : Argument[0] -> ReturnValue`

I might be overthinking this, but `Argument[0] -> ReturnValue` didn't quite make sense to me for this one since `Argument[0]` is a key, and the `ReturnValue` should be the value that the key maps to (based on my understanding of the [docs](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#getProperty-java.lang.String-)), so I was trying to take that mapping into account. Is taking the mapping into account not necessary for this model?

Reply:
Yes, propagating the taint from the argument to the return value isn't correct because controlling the key doesn't imply controlling the value stored with that key in the system properties.

Since a static method doesn't really have a useful qualifier, I think you'll need to use [synthetic globals](https://github.com/github/codeql/pull/10814/) for this. See the test for an example with arrays (would need to be adapted to maps):

```ql
"my.qltest.synth;A;false;storeInArray;(String);;Argument[0];SyntheticGlobal[db1].ArrayElement;value;manual",
"my.qltest.synth;A;false;readArray;();;SyntheticGlobal[db1].ArrayElement;ReturnValue;value;manual",
```

Now if you only want to create edges when the key used to set the property matches the key used to read the property, then I think MaD isn't enough and you'll need to use regular CodeQL (probably an `AdditionalValueStep`, or synthetic callables + synthetic globals for more advanced cases). Which is (probably?) out of the scope of this PR.

Reply:
Thanks! Updated in https://github.com/github/codeql/pull/11779/commits/0c7ffb055471c116ed26e0e244b3b517aa61d4ce. Let me know if it does not look correct.

> Which is (probably?) out of the scope of this PR.

I agree that that sounds out of scope for this PR.

Reply:
We should keep an eye on this in DCA, it may cause a lot of FP flow from a `set` of a property to a `get` of a different one.

Reply:
See https://github.com/github/codeql/pull/11779#discussion_r1071233634